packages/code-du-travail-modeles/src/__test__/depart.mise.retraite.commerce.detail.fruits.legumes.spec.ts

```diff
@@ -0,0 +1,137 @@
+    `(
+      "Pour un salarié de tel $category possédant $seniority mois d'ancienneté, son préavis devrait être $expectedResult mois",
+      ({ seniority, category, expectedResult, expectedReferences }) => {
+        const situation = engine.setSituation({
+          "contrat salarié . convention collective": "'IDCC1505'",
+          "contrat salarié . mise à la retraite": "oui",
+          "contrat salarié . travailleur handicapé": "non",
+          "contrat salarié . ancienneté": seniority,
+          "contrat salarié . convention collective . commerces de détail fruits et légumes . catégorie professionnelle": `'${category}'`,
+        });
+
+        const result = situation.evaluate(
+          "contrat salarié . préavis de retraite"
+        );
+        const references = getReferences(situation);
+        expect(result.missingVariables).toEqual({});
+        expect(result.nodeValue).toEqual(expectedResult);
+        expect(result.unit?.numerators).toEqual(["mois"]);
+        expect(references).toHaveLength(expectedReferences.length);
+        expect(references).toEqual(expect.arrayContaining(expectedReferences));
+      }
+    );
+  });
+
+  describe("Notifications", () => {
+    const notification =
+      "Attention: L'article de la convention collective ou la convention collective saisie n’a pas été étendue au niveau national. Par conséquent, pour que ce résultat soit applicable à votre situation, il faut que l’employeur ait adhéré à l’organisation patronale signataire de cette convention. Sans cette adhésion, l'employeur n'a pas l'obligation d'appliquer les règles de la convention mais il applique le préavis prévu par le code du travail.";
+    test.each`
+      seniority | category                                   | expectedNotificationLength | expectedNotification
+      ${5}      | ${"Ouvriers, Employés (Niveaux N1 à N4)"}  | ${0}                       | ${null}
+      ${6}      | ${"Ouvriers, Employés (Niveaux N1 à N4)"}  | ${0}                       | ${null}
+      ${24}     | ${"Ouvriers, Employés (Niveaux N1 à N4)"}  | ${0}                       | ${null}
+      ${5}      | ${"Agents de maîtrise (Niveaux N5 et N6)"} | ${0}                       | ${null}
+      ${6}      | ${"Agents de maîtrise (Niveaux N5 et N6)"} | ${0}                       | ${null}
+      ${24}     | ${"Agents de maîtrise (Niveaux N5 et N6)"} | ${0}                       | ${null}
+      ${5}      | ${"Cadres (Niveaux N7 et N8)"}             | ${1}                       | ${notification}
+      ${6}      | ${"Cadres (Niveaux N7 et N8)"}             | ${1}                       | ${notification}
+      ${24}     | ${"Cadres (Niveaux N7 et N8)"}             | ${1}                       | ${notification}
+    `(
+      "Pour un $category possédant $seniority mois d'ancienneté en départ à la retraite, le nombre de notification a affiché est de $expectedNotificationLength",
```

Comment:
On peut simplifier d'après la doc jest, tu peux appeler un attribut sur un élément:
```suggestion
      "Pour un $category possédant $seniority mois d'ancienneté en départ à la retraite, le nombre de notification a affiché est de $expectedNotification.length",
```

source : https://jestjs.io/docs/api#testeachtablename-fn-timeout

Reply:
Ok, oui en effet, il faut juste mettre en tableau, et je pense que c'est mieux. En effet 😄